src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.impl.operation.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler =
+                    operationStrategy.getOperationHandler(transaction.getOperation());
+
+            if (handler == null) {
+                throw new IllegalArgumentException(
+                        "No handler found for operation: " + transaction.getOperation()
+                );
```

Comment:
The check for a null `handler` is good for ensuring that a valid operation handler is available. However, consider logging the error or providing more detailed information about the transaction that caused the issue to aid in debugging.

Reply:
The error is descriptive enough